docs/revsets.md

```diff
@@ -120,32 +124,39 @@ revsets (expressions) as arguments.
 * `latest(x[, count])`: Latest `count` commits in `x`, based on committer
   timestamp. The default `count` is 1.
 * `merges()`: Merge commits.
-* `description(pattern)`: Commits with the given string in their
-  description.
-* `author(pattern)`: Commits with the given string in the author's name or
-  email.
+* `description(pattern)`: Commits that have a description matching the given
+  [string pattern](#string-patterns).
+* `author(pattern)`: Commits with the author's name or email matching the given
+  [string pattern](#string-patterns).
 * `mine()`: Commits where the author's email matches the email of the current
   user.
-* `committer(pattern)`: Commits with the given string in the committer's
-  name or email.
+* `committer(pattern)`: Commits with the committer's  name or email matching the
+given [string pattern](#string-patterns).
 * `empty()`: Commits modifying no files. This also includes `merges()` without
   user modifications and `root()`.
-* `file(pattern..)`: Commits modifying the paths specified by the `pattern..`.
-  Paths are relative to the directory `jj` was invoked from. A directory name
-  will match all files in that directory and its subdirectories. For example,
-  `file(foo)` will match files `foo`, `foo/bar`, `foo/bar/baz`, but not file
-  `foobar`.
+
+*   `file(string)` or `file("string"[, "string"]...)`: Commits modifying the
+    paths specified by one of the given strings. Currently, string patterns are
+    *not* supported. Paths are relative to the directory `jj` was invoked from.
```

Comment:
maybe better to say "paths", not "strings" or "string pattern".

We'll probably add `glob:` syntax, but it'll be different from the glob string pattern (since "*" shouldn't match "/" for example.) File patterns will have different set of prefixes.
https://repo.mercurial-scm.org/hg/help/patterns

Reply:
In the page that I linked, https://docs.rs/glob/latest/glob/struct.Pattern.html, those patterns seem intended for paths, as they explicitly support `**` that only makes sense there. Is there a reason not to use that? Performance, perhaps?

Reply:
Re name: I'm not sure, these are path fragments, certainly not absolute paths. Maybe "relative path" instead of "string"? It's longer, but much more describe.

Reply:
> In the page that I linked, https://docs.rs/glob/latest/glob/struct.Pattern.html, those patterns seem intended for paths, as they explicitly support `**` that only makes sense there. Is there a reason not to use that? Performance, perhaps?

Well, the glob crate could be used under the hood, but the evaluation rule would be different. `description(glob:"a*b")` will match "alice/bob" for example. Path patterns will have concepts of relative-to-repo, relative-to-cwd, etc.

> Re name: I'm not sure, these are path fragments, certainly not absolute paths. Maybe "relative path" instead of "string"? It's longer, but much more describe.

I don't have good idea, but maybe we can say "path pattern"? I think the important thing to be mentioned here is that `pattern` in `file(pattern..)` is NOT a "string pattern".

Reply:
I agree that we should at least avoid using "string pattern". We could use "path prefixes" for now since that's all we support.

Reply:
How about `path_prefix`?

Reply:
`path_prefix` sounds like a symbol in some language (e.g. the revset or template language). I'd prefer "path prefix". I'm worried users will go searching our docs and source code for `path_prefix` if we use that.

Reply:
`file(path prefix)` makes it seem like there might be two things. So, I changed it to `file(pathprefix)`.

Reply:
Actually, I prefer `relativepath`, since `file(foo)` doesn't match `foobar`.

Reply:
Ah, I see. Yeah, I agree then. (Sorry, I didn't check where we were talking about this and thought it was in prose.)

> Actually, I prefer `relativepath`, since `file(foo)` doesn't match `foobar`.

I think it's quite common for "path prefix" `foo` to not match `foobar`. At least there's one more example: https://doc.rust-lang.org/stable/std/path/struct.Path.html#method.strip_prefix :)

Reply:
I rewrote it with "relative path", and I like the result. I think Yuya's original suggestion of "path" would've been OK too.

In particular, I'm glad to be rid of the word "prefix" that I previously introduced here. It was based on my original incorrect understanding of what this does. While it doesn't necessarily imply that "foo" must match "foobar", it does leave it quite ambiguous.